src/overlays/actors/ovl_En_Tab/z_en_tab.c

```diff
@@ -31,75 +65,552 @@ const ActorInit En_Tab_InitVars = {
+                    tempActor = &GET_PLAYER(globalCtx)->actor;
+                    this->actor.child = tempActor;
+                } else {
+                    tempActor = &this->unk_1E4->actor;
+                    this->actor.child = tempActor;
+                }
+
+                if (func_80BE0C04(this, tempActor, 160.0f)) {
+                    this->unk_2FC |= 8;
+                } else {
+                    this->unk_2FC &= ~8;
+                }
+            }
+            break;
+
+        case 2:
+            if (func_80BE0C04(this, &GET_PLAYER(globalCtx)->actor, 200.0f)) {
+                this->unk_2FC |= 8;
+            } else {
+                this->unk_2FC &= ~8;
+            }
+            break;
+    }
+
+    return false;
+}
+
+void func_80BE1224(EnTab* this, GlobalContext* globalCtx) {
+    if ((this->unk_1D8 == 1) || (this->unk_1D8 == 2)) {
+        func_80BE10BC(this, globalCtx);
+    }
+    Math_ApproachS(&this->actor.shape.rot.y, this->actor.world.rot.y, 3, 0x2AA8);
+}
+
+void func_80BE127C(EnTab* this, GlobalContext* globalCtx) {
+    u32* unk_14 = &gSaveContext.unk_14;
+    struct_80133038_arg2 sp18;
+
+    this->unk_31A = REG(15) + *unk_14;
+
+    if (!func_80133038(globalCtx, D_80BE18D0, &sp18) ||
+        ((this->unk_1D8 != sp18.unk0) && !func_80BE1060(this, globalCtx, &sp18))) {
+        this->actor.shape.shadowDraw = NULL;
+        this->actor.flags &= ~ACTOR_FLAG_1;
+        sp18.unk0 = 0;
+    } else {
+        this->actor.shape.shadowDraw = ActorShadow_DrawCircle;
+        this->actor.flags |= ACTOR_FLAG_1;
+    }
+    this->unk_1D8 = sp18.unk0;
+    func_80BE1224(this, globalCtx);
+}
+
+void func_80BE1348(EnTab* this, GlobalContext* globalCtx) {
+    s32 pad;
+    Vec3f sp40;
+    Vec3f sp34;
+
+    if (func_8010BF58(&this->actor, globalCtx, func_80BE0E04(this, globalCtx), this->unk_328, &this->unk_1DC)) {
+        func_8013AED4(&this->unk_2FC, 3, 7);
+        this->unk_2FC &= ~8;
+        this->unk_2FC |= 0x40;
+        this->unk_324 = 20;
+        this->unk_1DC = 0;
+        this->actionFunc = func_80BE127C;
+    } else if (this->unk_1E0 != 0) {
+        Math_Vec3f_Copy(&sp40, &this->unk_1E0->world.pos);
+        Math_Vec3f_Copy(&sp34, &this->actor.world.pos);
+        Math_ApproachS(&this->actor.shape.rot.y, Math_Vec3f_Yaw(&sp34, &sp40), 4, 0x2AA8);
+    }
+}
+
+void EnTab_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnTab* this = THIS;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 14.0f);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &object_tab_Skel_007F78, NULL, this->jointTable, this->morphTable,
+                       20);
+    this->unk_32C = -1;
+    func_80BE05BC(this, 0);
+    Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, DamageTable_Get(0x16), &sColChkInfoInit);
```

Comment:
Musing: This is confusing but is also present in multiple other decompiled dialogue heavy actors:

```
CollisionCheck_SetInfo2(&this->actor.colChkInfo, DamageTable_Get(0x16), &sColChkInfoInit);
```
Why would a dialogue NPC like this have a DamageTable?